### DISCUSSION TITLE
Use stomp to handle websocket messaging

### DISCUSSION THREAD

**Discussion by wnleao** [sevdokimov/log-viewer#30] [Ideas]
What do you think about using STOMP to handle to messaging protocol? Have you considered it before? If you have, why didn't you go with it? Is there any issue that I might be overlooking?

https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#websocket-stomp
https://programming.vip/docs/four-ways-of-integrating-websocket-with-spring-boot.html
https://spring.io/guides/gs/messaging-stomp-websocket/

**Comment by sevdokimov** [comment]
> Have you considered it before?

No, I am not familiar with STOMP. 
Log viewer can be embedded into another Java Web application as a library, so it must have as few dependencies as possible to avoid conflicts. And it should be as light as possible. 
Also, it should be able to work without WebSocket. I experienced the situation where WebSocket is not supported by a proxy.

Reinventing the wheel is not a best practice, but I created a custom wrapper under websocket/plain_http. See `LogSession` class, the methods with @Remote annotation can be invoked by UI. On UI side a method call looks like:
```
                this.commService.send(
                    new Command('initPermalink', {
                        recordCount: argValue1,
                        linkHash: argValue1,
                    })
                );
```

`LogViewComponent` UI component contains methods with @BackendEventHandler() annotation, which are calling when backend sends messages to UI.
Let's keep it as is.

**Reply by wnleao** [reply]
Hey! Yeah, I've studied those. I think that the project could benefit from implementing/using stomp. During the week as I find time in my schedule I could setup a small example to test whether it would work and how that code would look like.
